### PR TITLE
Sils/spacing

### PR DESCRIPTION
**Pull request by sils1297**
Hi, I saw https://github.com/duckduckgo/zeroclickinfo-goodies/pull/2288#discussion_r52844722 and thought I might help since we're currently developing a tool that can do all kinds of static code analysis.

This PR corrects all the spacing youve got globally on all unicode files. It also adds a config (if you don't want it I can of course remove the commit) for the tool I used. With the config you can use gitmate.io which is the free software service I'm currently building, it's a prototype and not very reliable but it'll automatically tell you if new issues arise as PR comments. (You *can* use it in production but you really shouldn't rely on it and for sure first test if you like all the features on a test repo since you can't turn anything off yet.) If that sounds too shaky (I bet it does) you can just invoke `coala-ci` in your CI builds and get some pretty output if something is not according to the defined style.

While I was at it I spotted some JSON and unified the spacing there too (if you want it can be sorted as well :)). If you want deeper analysis there's lots of it. CSS, JS, Python, you name it. (If it's missing we can add it.)

### CODE REVIEW

**Comment by GuiltyDolphin** [comment]
@sils1297 My word ;)

Thanks for this - I'm just putting it on hold for the time being so we can get some people to have a look at it. The main issue is that this is likely to cause *a lot* of conflicts with existing pull requests! :eyes:

/cc @moollaza @jagtalon @duckduckgo/community-leaders

**Comment by sils1297** [comment]
Sure no problem. I have to add the disclaimer that I'm the maintainer of that little tool but we have honest intentions and no commercial interest (actually mostly freetime working university students.)

I could change the PR to only add the config file in one commit and do a second that just corrects everything so whenever a conflict arises we just redo the second commit?

**Comment by GuiltyDolphin** [comment]
@sils1297 Humm.. Would make more sense - but I'd wait for one of the others to arrive, I think their voices would be best with something of this size.

**Comment by GuiltyDolphin** [comment]
@sils1297 By the way, #2281 deals with quite a bit of this already - most of spacing issues are with cheat sheets (we get more cheat sheets than other IAs), and that introduces various tests, including some for code style.

**Comment by sils1297** [comment]
Alright I'll let you figure out what you want. coala will probably scale better and provides neat UIs including editor plugins and stuff, it allows you to do all kinds of analysis. Your solution OTOH doesn't need a dependency to run.

**Comment by moollaza** [comment]
@sils1297 thanks for this. I'll discuss it with the team and get back to you.

One thing I noticed is that all the UTF-8 characters have been replaced in our JSON files -- I'd like for them to stay as-is. Having them in the text makes it much easier to read/understand/modify (especially accented characters in the middle of a word).

**Comment by moollaza** [comment]
FWIW I'm most interested in the automated reporting of bugs, so that we don't have to add them ourselves. I've previously investigated https://houndci.com/ but it has no support for Perl or JSON (I think?)

**Comment by GuiltyDolphin** [comment]
@moollaza Yeah, I can think of a few, but none of them support Perl out-of-the-box AFAIK.

**Comment by sils1297** [comment]
For coverage I'd recommend codecov.io. We've made good experiences with that.

We have a bug about the unicode thing, we'll look into getting that fixed ASAP.

We have a PerlCriticBear which does analysis on Perl code out of the box. You just need to install perlcritic for it to work and then you can configure and use anything seamlessly via coala without having to learn every of those tools. We can add more linters as per your demand.

coala-ci will give you an output like this: https://twitter.com/coala_analyzer/status/698106462434959360 so it should be obvious for contributors what to do.

**Comment by sils1297** [comment]
Ah the JSON unicode thing seems to be something else, filed a bug at https://github.com/coala-analyzer/coala/issues/1527.

**Comment by GuiltyDolphin** [comment]
@moollaza Looks like we've already [been using Coveralls](https://coveralls.io/github/duckduckgo/zeroclickinfo-spice) for Spice?

**Comment by sils1297** [comment]
I took a minute to add a parameter so you can keep your unicode chars, it'll be in the next development release of coala today or tomorrow. https://github.com/coala-analyzer/coala/pull/1541

**Comment by sils1297** [comment]
@moollaza the unicode characters are now as they are, in addition if there were any encoded characters before they are converted to unicode chars by coala. Please note that in order to achieve those results, the development version of coala (`pip3 install coala --pre`) needs to be used since I just added that option.

File: lib/DDG/Goodie/SumOfNaturalNumbers.pm [hunk 1]
```diff
@@ -28,10 +28,10 @@ handle remainder => sub {
     return unless $_ =~ /^(?:from )?(\d+)\s*(to|-)\s*(\d+)$/i;
 
     if ($1 > $3) {
-	return;
+    return;
```
Comment (moollaza) [inline lib/DDG/Goodie/SumOfNaturalNumbers.pm]: Looks like this one needs manual fixing to correct the indentation

**Comment by moollaza** [comment]
@sils1297 apologies for the delay here, and thanks again for all the hard work you've put into improving this Goodies Repo. You've done a great job so far! We at DuckDuckGo, and the DuckDuckHack community really appreciate you taking the time to contribute.

It looks like this still needs some work? I see there are merge conflicts again. To be honest though, I'm still a worried about merging this because we will end up creating merge-conflicts for every open PR. As mentioned though, I'm really interested in the CI approach, especially if we could have your bot comment on PR's to point out style breakage. How close is that functionality to completion?

I'd love to chat if you're around this week or next. Please message me on slack (@moollaza) or send me an email (moollaza@duckduckgo.com) if you'd be interested in talking. 

Thanks again!

**Comment by sils1297** [comment]
@moollaza you're welcome, applying coala to projects and helping them is awesome after we're putting so much work into it :)

You're in luck, our CI service has just reached beta status and we don't expect to do evil things like wiping all data anymore. I'd recommend you try https://app.gitmate.io/ on some example repository. GitMate will always stay free for public repositories and if the service doesn't work out (i.e. if I won't be able to keep the servers running financially) we'll release the source code as free software which we might do anyway. We're using GitMate for coala development quite productively for some time now.

You can always reach me and our communities on coala related matters at https://gitter.im/coala-analyzer/coala, on GitMate (the CI service) related matters on https://gitter.im/GitMateIO/support or directly via lasse.schuirmann@gmail.com.

**Comment by sils1297** [comment]
FWIW since this is on hold I won't resolve the conflicts now but it's really easy to do that, just reapply coala on the files and everything will be fixed magically ;)

**Comment by mintsoft** [comment]
@moollaza @GuiltyDolphin Any idea what we're going to do with this one?

**Comment by sils1297** [comment]
FWIW you could just take the configuration and activate GitMate, then whenever a line is changed GitMate will raise a concern as a review comment in a pull request. Thus you can avoid correcting old code and only correct code when it changes.

**Comment by GuiltyDolphin** [comment]
@sils1297 @mintsoft IDK... It'd be nice to have code climate running, but in terms of cheat sheet styling we should be moving away from local JSON entirely in the future if all goes well.

So many conflicts ^.^

**Comment by mintsoft** [comment]
@moollaza @sils1297 @GuiltyDolphin I'm going to close this for now; any objections; feel free to reopen

**Pull request closed without merging**